felix/fv/service_loop_prevention_test.go

```diff
@@ -223,20 +199,14 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ service loop prevention; wi
 		tryRoutingLoop(false, 1)
 
 		By("configuring ServiceLoopPrevention=Reject")
-		updateFelixConfig(func(cfg *api.FelixConfiguration) {
-			cfg.Spec.ServiceLoopPrevention = "Reject"
-		})
+		setSvcLoopPrevention(tc, "Reject")
 
 		// Expect to see rules in cali-cidr-block chains with REJECT.  (Allowing time for a
 		// Felix restart.)
-		for id, felix := range tc.Felixes {
-			if BPFMode() {
-				curId := tcProgID[id]
-				Eventually(func() int {
-					tcProgID[id] = getTcProgID(felix, "eth0")
-					return tcProgID[id]
-				}, "15s", "1s").ShouldNot(BeNumerically("==", curId))
-			} else {
+		if BPFMode() {
+			ensureAllNodesBPFProgramsAttached(tc.Felixes)
```

Comment:
is this enough? Shoudn't we also check for the right routes to be programmed?

Reply:
The routes won't change as there is no change in the BGPConfig. Felix config changes. Hence it is enough if we wait for the programs to be attached after felix restart.